### ISSUE TITLE
Large matrices produce error

### ISSUE THREAD

**Issue by NickCH-K** [haghish/rcall#12]
I've noticed that creating a matrix at least a certain size (not necessarily that big, certainly not too big for Stata's matrices) produces an error on the Stata side. The matrix is produced fine in R, but then it has trouble being sent back to Stata, which produces an error. This can be a real problem if you want to create a matrix as an intermediate step to something in R, or want to create a new variable in R and ship it back to Stata in matrix form for `svmat`.

```stata
sysuse auto.dta, clear

* Error:
rcall clear
rcall: df <- st.data(); test <- model.matrix(lm(price ~ -1 + ., data = df))
* test not found, although test created properly
* rcall will continue to throw "test not found" error on every call until cleared
rcall: summary(test)

* Also error:
rcall clear
rcall: M <- matrix(1:900, nrow = 30)
```

**Comment by haghish** [comment]
That is true and I was planning to fix this problem back in 2016! Stata applies a limit for iterating on matrices (or local's length, not certain atm) within an ado-program (at least on Stata 13 and 14). The error is given by Stata when the limit is reached. 

For reference, a quick solution can be done using a user-written package that saves and Stata matrices to CSV (or binary). I remember that back in 2016 such a package existed and I wanted to use it to update the package...

**Comment by NickCH-K** [comment]
Good to know! I suppose for now I'll find a workaround.

Would you say there's a best-practice `rcall` way to get a newly created variable from R back into Stata when it's not the same length as the original data? Seems valuable if you're using R to create new variables, since R handles NAs differently. For example:

```
preserve
keep if condition
rcall: df <- st.data() ; newvar <- predict(lm(y~x, data=df))
restore
* put newvar into the Stata data here
```

(imagine I can't do the subsetting on the R side)

Thus far I've been doing this either by sending it back as a local and reading it into Stata data by looping over it, or sending it back as a matrix and using `svmat` on the subsetted data, but both seem problematic for bigger data sets.

Currently thinking of doing something like

```
g identifier = _n
save origdata.dta, replace
keep if condition
rcall: df <- st.data() ; df$newvar <- predict(lm(y~x, data=df)) ; df <- df[,c('identifier','newvar')] ; st.load(df)
merge 1:1 identifier using origdata.dta
drop _merge identifier
erase origdata.dta
```

But is there a better way?

**Comment by haghish** [comment]
Stata returns an error for any matrix with more than 300 cells. 

So:

    rcall: M <- matrix(1:300, nrow = 30)  // would work

    rcall: M <- matrix(1:301, nrow = 30)  //produces the error you mentioned

I will automate saving the large matrices both direction, from Stata to R and R to Stata in version 3.0

**Comment by haghish** [comment]
solved in version 3.0.0 BETA
